modules/unsupported/geojson-core/src/main/java/org/geotools/filter/expression/JSONNodePropertyAccessorFactory.java

```diff
@@ -0,0 +1,212 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2022, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.expression;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.Date;
+import java.util.Map;
+import org.geotools.data.geojson.GeoJSONReader;
+import org.geotools.util.factory.Hints;
+import org.opengis.feature.Property;
+import org.opengis.feature.simple.SimpleFeature;
+
+/** Property Accessor Factory for {@link JsonNode} */
+public class JSONNodePropertyAccessorFactory implements PropertyAccessorFactory {
+
+    static PropertyAccessor JSONNODEPROPERTY = new JSONNodePropertyAccessor();
+    static final String ROOT = "/";
+
+    @Override
+    public PropertyAccessor createPropertyAccessor(
+            Class<?> type, String xpath, Class<?> target, Hints hints) {
+        if (xpath == null) {
```

Comment:
Should you check if "type" is a SimpleFeature? See SimpleFeaturePropertyAccessorFactory.

Reply:
Added